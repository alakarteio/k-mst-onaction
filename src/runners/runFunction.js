import { isArray } from 'lodash'

export default (dispatch, action, tree) => {
  const dispatchCall = dispatch(action, tree)
  if (!dispatchCall) return
  if (isArray(dispatchCall)) dispatchCall.forEach(runner => runner(action, tree))
}
