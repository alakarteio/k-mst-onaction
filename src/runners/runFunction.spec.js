/* eslint-env jest */
import runFunction from './runFunction'

describe('function runner', () => {
  it('should run a simple function', () => {
    const dispatch = jest.fn(() => 'dispatch function')

    runFunction(dispatch, { ho: 'i am an action' }, { this: 'is the tree :)' })

    expect(dispatch.mock.calls).toMatchSnapshot()
  })

  it('should run a simple function that returns nothing', () => {
    const dispatch = jest.fn()

    runFunction(dispatch, { ho: 'i am an action' }, { this: 'is the tree :)' })

    expect(dispatch.mock.calls).toMatchSnapshot()
  })

  it('should run a function that returns an array', () => {
    const first = jest.fn(() => 'first')
    const second = jest.fn(() => 'second')
    const dispatch = jest.fn(() => [first, second])

    runFunction(dispatch, { ho: 'i am an action' }, { this: 'is the tree :)' })

    expect(dispatch.mock.calls).toMatchSnapshot()
    expect(first.mock.calls).toMatchSnapshot()
    expect(second.mock.calls).toMatchSnapshot()
  })
})
