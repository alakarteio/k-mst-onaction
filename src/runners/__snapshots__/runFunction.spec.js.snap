// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`function runner should run a function that returns an array 1`] = `
Array [
  Array [
    Object {
      "ho": "i am an action",
    },
    Object {
      "this": "is the tree :)",
    },
  ],
]
`;

exports[`function runner should run a function that returns an array 2`] = `
Array [
  Array [
    Object {
      "ho": "i am an action",
    },
    Object {
      "this": "is the tree :)",
    },
  ],
]
`;

exports[`function runner should run a function that returns an array 3`] = `
Array [
  Array [
    Object {
      "ho": "i am an action",
    },
    Object {
      "this": "is the tree :)",
    },
  ],
]
`;

exports[`function runner should run a simple function 1`] = `
Array [
  Array [
    Object {
      "ho": "i am an action",
    },
    Object {
      "this": "is the tree :)",
    },
  ],
]
`;

exports[`function runner should run a simple function that returns nothing 1`] = `
Array [
  Array [
    Object {
      "ho": "i am an action",
    },
    Object {
      "this": "is the tree :)",
    },
  ],
]
`;
